nemo/collections/nlp/parts/nlp_overrides.py

```diff
@@ -584,6 +589,51 @@ def _integrate_original_checkpoint_data(self, checkpoint: Dict[str, Any]) -> Dic
 
         return checkpoint
 
+    def _drop_optimizer_states(self, filepath: Union[str, Path], storage_options: Optional[Any] = None) -> None:
+        # Get list of saved checkpoints
+        checkpoints = self._get_checkpoints_list(filepath)
+
+        # Drop optimizer states
+        checkpoint_index = len(checkpoints) - self.save_last_n_optim_states - 1
+        if len(checkpoints) > self.save_last_n_optim_states:
+            checkpoint_path = checkpoints[checkpoint_index]
+
+            logging.info(f"Loading '{checkpoint_path}' checkpoint to drop optimizer states...")
+            checkpoint = self.load_checkpoint(checkpoint_path=checkpoint_path)
+
+            # Remove the checkpoint version with optimizer states
+            self.remove_checkpoint(checkpoint_path)
```

Comment:
@mikolajblaz but in this case we'll get an error that checkpoint exitst and not empty or should we change the name for checkpoint w/o optimizer states?

Reply:
We probably have to save to a temp dir and rename.
Imagine the following case: at iter 1000 we have 3 checkpoints from iters 800, 900, 1000 and we want to remove the optimizer state from iter 80. If during removal we have an error we lose the iter 800 checkpoint forever (we can't even retrain the model for a while) and it could be e.g. checkpoint with Top1 val_loss